be/src/service/service_be/starrocks_be.cpp

```diff
@@ -89,29 +87,68 @@ Status init_datacache(GlobalEnv* global_env, const std::vector<StorePath>& stora
         }
         RETURN_IF_ERROR(DataCacheUtils::parse_conf_datacache_mem_size(config::datacache_mem_size, mem_limit,
                                                                       &cache_options.mem_space_size));
-        if (config::datacache_disk_path.value().empty()) {
-            // If the disk cache does not be configured for datacache, set default path according storage path.
-            std::vector<std::string> datacache_paths;
-            std::for_each(storage_paths.begin(), storage_paths.end(), [&](const StorePath& root_path) {
-                datacache_paths.push_back(root_path.path + "/datacache");
-                // Clear the residual datacache files
-                std::filesystem::path sp(root_path.path);
-                auto old_path = sp.parent_path() / "datacache";
-                DataCacheUtils::clean_residual_datacache(old_path.string());
-            });
-            config::datacache_disk_path = JoinStrings(datacache_paths, ";");
-        }
-        RETURN_IF_ERROR(DataCacheUtils::parse_conf_datacache_disk_spaces(
-                config::datacache_disk_path, config::datacache_disk_size, config::ignore_broken_disk,
-                &cache_options.disk_spaces));
 
-        size_t total_quota_byts = 0;
-        for (auto& space : cache_options.disk_spaces) {
-            total_quota_byts += space.size;
+        size_t total_quota_bytes = 0;
+        for (auto& root_path : storage_paths) {
+            // Because we have unified the datacache between datalake and starlet, we also need to unify the
+            // cache path and quota.
+            // To reuse the old cache data in `starlet_cache` directory, we try to rename it to the new `datacache`
+            // directory if it exists. To avoid the risk of cross disk renaming of a large amount of cached data,
+            // we do not automatically rename it when the source and destination directories are on different disks.
+            // In this case, users should manually remount the directories and restart them.
+            std::filesystem::path datacache_path(root_path.path + "/datacache");
+#ifdef USE_STAROS
+            if (config::datacache_unified_instance) {
+                std::filesystem::path starlet_cache_path(root_path.path + "/starlet_cache/star_cache");
+                if (std::filesystem::exists(starlet_cache_path)) {
+                    if (DiskInfo::disk_id(starlet_cache_path.c_str()) != DiskInfo::disk_id(datacache_path.c_str())) {
+                        LOG(ERROR) << "The datacache directory and the old starlet_cache directory cannot be located on different disks. "
+                                   << "Please manually mount the datacache to the same disk as starlet_cache and then restart again";
+                        return Status::InternalError("The datacache directory is different with old starlet_cache directory");
```

Comment:
> may be do not return error and set `datacache_unified_instance` to `false`, so that user still can start be?
This configuration item is mainly designed to solve the problem of unified instance exceptions, and users configure to start individual instances for fault tolerance. So this situation is usually more appropriate for users to know and make their own choices. Automatically switching to independent mode may cause unexpected problems due to resource usage and other factors.